ui/backend/client_ui.py

```diff
@@ -131,19 +131,23 @@ def update_similar_snippets(self,
             The number of similar snippets found and updated
         """
 
-        discoveries = self.get_discoveries(repo_url,
-                                           file_name,
-                                           state_filter='new')[1]
-        model = build_embedding_model()
-        target_snippet_embedding = compute_snippet_embedding(target_snippet,
-                                                             model)
+        discoveries = self.get_discoveries(repo_url, file_name)[1]
+        # Compute target snippet embedding
+        target_embedding = self.get_embedding(snippet=target_snippet)
         n_updated_snippets = 0
-        for d in discoveries:
-            snippet_embedding = compute_snippet_embedding(d['snippet'],
-                                                          model)
-            similarity = compute_similarity(target_snippet_embedding,
-                                            snippet_embedding)
-            if similarity > threshold:
-                n_updated_snippets += 1
-                self.update_discovery(d['id'], state)
-        return n_updated_snippets
+        if not target_embedding:
+            return 0
+        else:
+            for d in discoveries:
+                if (
+                    d['state'] != state
+                    and self.get_embedding(discovery_id=d['id'])
+                ):
+                    # Compute similarity of target embedding and embedding
+                    embedding = self.get_embedding(discovery_id=d['id'])
+                    similarity = compute_similarity(target_embedding,
+                                                    embedding)
+                    if similarity > threshold:
+                        n_updated_snippets += 1
+                        self.update_discovery(d['id'], state)
+            return n_updated_snippets
```

Comment:
isn't this block of code same as in the parent class?

Reply:
yes this block is the same as in the parent class, but unlike the entire method, so I cannot simply delete the method from the UI class